Doc/library/shutil.rst

```diff
@@ -209,14 +209,19 @@ Directory and files operations
 
 
 .. function:: copytree(src, dst, symlinks=False, ignore=None, \
-              copy_function=copy2, ignore_dangling_symlinks=False)
+              copy_function=copy2, ignore_dangling_symlinks=False, \
+              dirs_exist_ok=False)
 
    Recursively copy an entire directory tree rooted at *src*, returning the
-   destination directory.  The destination
-   directory, named by *dst*, must not already exist; it will be created as
-   well as missing parent directories.  Permissions and times of directories
-   are copied with :func:`copystat`, individual files are copied using
-   :func:`shutil.copy2`.
+   destination directory. If *dirs_exist_ok* is false (the default), then the
+   destination directory, named by *dst*, must not already exist; it will be
+   created as well as missing parent directories.  Permissions and times of
+   directories are copied with :func:`copystat`, individual files are copied
+   using :func:`shutil.copy2`.
+
+   If *dirs_exist_ok* is true, then the destination directory may already exist,
+   and existing files in the destination tree will be overwritten by matching
+   files in the source tree.
```

Comment:
I would simply rephrase it as something along these lines:
```rst
*dirs_exist_ok* dictates whether to raise an exception in case *dst* as well as 
any missing parent directory already exists. 
```
I would not talk about the overwriting of files because the argument specifically targets directories and has nothing to do with files. When copying a big directory another process may create files in the *dst* directory which conflicts with the ones living in `src`, and they are always overwritten even when `dirs_exist_ok=False`.

Reply:
The behavior of `dirs_exist_ok=False` (raising an exception) is already documented in the previous paragraph, so your rephrasing of this paragraph seems redundant. The purpose of this paragraph was to document the behavior of `dirs_exist_ok=True`.

I've just removed the part that talks about files.

Reply:
@giampaolo good to merge now?